Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -427,6 +432,9 @@ private Branch(@NonNull Context context) {
         if (!trackingController.isTrackingDisabled()) { // Do not get GAID when tracking is disabled
             isGAParamsFetchInProgress_ = deviceInfo_.getSystemObserver().prefetchAdsParams(context,this);
         }
+        if (prefHelper_.isAutoLogInAppPurchasesAsEventsEnabled()) {
+            new BillingGooglePlay().createBillingClient(context_, aBoolean -> null);
```

Comment:
Each time you call this a new instance is built from the builder. I recommend using a singleton pattern https://stackoverflow.com/questions/51834996/singleton-class-in-kotlin